src/main/java/net/minecraftforge/client/SkyRenderPass.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client;
+
+public enum SkyRenderPass
+{
+    /**
+     * Background sky.
+     * */
+    SKY_BACK,
+    /**
+     * Sun, moon, stars and others.
+     * */
+    SUN_MOON_STARS,
```

Comment:
Why are these combined into one pass?
Is it possible to draw over the stars and behind the moon (like a planet would)?
Is it possible to make this into a more general sorted system, so that mods can draw before or after any specific part?

Reply:
I tried to keep what vanilla does. And I only got the three pass as they are necessary in most cases. While the Sun, the Moon and Stars are optional and even don't exist.
And yes, it's pretty tricky to draw over the stars and behind the moon in this system. I think I can solve this by making the priority be changed independently.
By sorted system, you mean something like mod sorting system? I think there is high probability where a cycle could happen. Also it's much more complex, so I think it isn't worth it.